src/internal/hooks/use-visual-mode/index.ts

```diff
@@ -8,6 +8,9 @@ import { useMutationObserver } from '../use-mutation-observer';
 import { isDevelopment } from '../../is-development';
 import { warnOnce } from '../../logging';
 
+// Note that this hook doesn't take into consideration @media print (unlike the dark mode CSS),
```

Comment:
What was wrong with using `!window?.matchMedia?.('print').matches`?

Reply:
`window.matchMedia('print').matches` changes briefly to `true` and then immediately back to `false` when opening the print dialog, which doesn't give enough time for React to re-render.

`window.addEventListener('beforeprint')` (and `afterprint`) works better in some browsers but not consistently.

In general, the amount of & complexity of code that looked like it would be required for this seemed to outweigh the benefits of the usecase (it impacts CodeEditor and dropdown elements rendered in portals)

Reply:
I see, but what is the scope of the fix now? The `dark-mode-only` mixin is only used for icons in the Code editor component, and I am no able to find any difference by manually testing in Chrome (nor in the Code editor component pages nor elsewhere).

Reply:
All CSS-based dark mode styling should be removed when printing. However, I think the PR deployment isn't showing this up because the supporting change (https://github.com/cloudscape-design/theming-core/commit/5febd2e) has not yet made it through our internal pipelines